### PR TITLE
Run the unit tests again

### PR DESCRIPTION
**Pull request by mszabo-wikia**
In #5039, the unit tests were moved from tests/phpunit/Unit to directly under tests/phpunit, but the PHPUnit configuration was not updated to match. Due to this, 5k+ unit tests have not been running for ~3 years.

So, update the PHPUnit configuration to run these tests again and make fixes as needed. Due to the size of the changes, where nontrivial changes to non-test code would have been required to fix the tests, I opted to skip the test instead with a "SUT needs refactoring" message so that the necessary changes could be made later without complicating the review.

Many tests emit deprecation warnings on newer PHP. I only fixed missing property declarations in test code in this patch; the rest will need investigation to determine whether they are the result of incomplete or invalid stubbings or actual issues in the code.

### CODE REVIEW

File: src/Constraint/Constraints/ShapeConstraint.php [hunk 1]
```diff
@@ -181,7 +181,7 @@ private function hasMinLength( $minLength, $property ) {
 				continue;
 			}
 
-			if ( mb_strlen( $dataItem->getString() < $minLength ) ) {
```
Comment (mszabo-wikia) [inline src/Constraint/Constraints/ShapeConstraint.php:184]: 1/3 of trivial non-test code changes in this patch.

File: src/Localizer/LocalLanguage/LocalLanguage.php [hunk 2]
```diff
@@ -474,7 +474,7 @@ public function getPreferredDateFormatByPrecision( $precision = null ) {
 		);
 
 		foreach ( $dateOutputFormats as $key => $format ) {
-			if ( @constant( $key ) === $precision ) {
```
Comment (mszabo-wikia) [inline src/Localizer/LocalLanguage/LocalLanguage.php:477]: 2/3 of trivial non-test code changes in this patch.

File: src/Property/Annotators/EditProtectedPropertyAnnotator.php [hunk 3]
```diff
@@ -64,11 +64,6 @@ public function addTopIndicatorTo( ParserOutput $parserOutput ) {
 			return false;
 		}
 
-		// FIXME 3.0; Only MW 1.25+ (ParserOutput::setIndicator)
-		if ( !method_exists( $parserOutput, 'setIndicator' ) ) {
-			return false;
-		}
```
Comment (mszabo-wikia) [inline src/Property/Annotators/EditProtectedPropertyAnnotator.php:70]: 3/3 of trivial non-test code changes in this patch.

File: tests/phpunit/MediaWiki/Hooks/SpecialPageListTest.php
Comment (mszabo-wikia) [inline tests/phpunit/MediaWiki/Hooks/SpecialPageListTest.php:1]: This hook handler no longer exists.

File: tests/phpunit/ParserFunctions/SubobjectParserFunctionTest.php [hunk 4]
```diff
@@ -627,7 +627,7 @@ public function tokuFixedParameterProvider() {
 				],
 				'propertyValues' => [
 					'Foo Bar',
-					'1001'
```
Comment (mszabo-wikia) [inline tests/phpunit/ParserFunctions/SubobjectParserFunctionTest.php:630]: Updated expectations here and in annotator tests to account for a25c0263b376abe6858eaee627a18b8c189c3e58.

File: tests/phpunit/Integration/SpecialsTest.php [hunk 5]
```diff
@@ -38,7 +39,7 @@
  * @group SMWExtension
  * @group medium
  */
-class SpecialsTest extends SemanticMediaWikiTestCase {
```
Comment (mszabo-wikia) [inline tests/phpunit/Integration/SpecialsTest.php:41]: This test accesses the database and the service container, so it's an integration test - moving it accordingly.

**Comment by mszabo-wikia** [comment]
So for now I decided to run the unit tests separately in CI, as some of them seem to affect global state, causing test failures in other suites down the road. Also excluded it from < 1.38 as there are some additional failures there that need investigation.

**Comment by mszabo-wikia** [comment]
After some changes, the unit tests should now pass on all versions (when run separately)

File: src/MediaWiki/PageInfoProvider.php [hunk 6]
```diff
@@ -189,7 +189,7 @@ public static function isProtected( Title $title, string $action = '' ) {
 		}
 
 		// MW < 1.37
-		return $title->isProtected();
```
Comment (mszabo-wikia) [inline src/MediaWiki/PageInfoProvider.php:192]: 3/4 of trivial non-test code changes in this patch.

File: tests/phpunit/Constraint/Constraints/ShapeConstraintTest.php [hunk 7]
```diff
@@ -69,10 +70,10 @@ public function testCheckConstraint_shape_constraint_missing_property() {
 			->with( $this->equalTo( 'Foo' ) )
 			->will( $this->returnValue( false ) );
 
-		$dataValue = $this->getMockBuilder( '\SMWDataValue' )
-			->disableOriginalConstructor()
-			->setMethods( [ 'addError', 'getCallable' ] )
-			->getMockForAbstractClass();
+		$dataValue = $this->createMock( SMWDataValue::class );
+		$dataValue->expects( $this->any() )
```
Comment (Nikerabbit) [inline tests/phpunit/Constraint/Constraints/ShapeConstraintTest.php:74]: Isn't `->expects( $this->any() )` the default behaviour?

File: tests/phpunit/Elastic/Indexer/IndexerTest.php [hunk 8]
```diff
@@ -81,7 +82,6 @@ public function testCreate() {
 
 		$expected = [
 			'index' => '_index_abc',
-			'type' => 'data',
```
Comment (Nikerabbit) [inline tests/phpunit/Elastic/Indexer/IndexerTest.php:84]: This change is not obvious to me.

**Comment by Nikerabbit** [comment]
Very nice!

File: tests/phpunit/Elastic/Indexer/IndexerTest.php (hunk 8 above)
Comment (mszabo-wikia) [reply tests/phpunit/Elastic/Indexer/IndexerTest.php:84]: This is to account for https://github.com/SemanticMediaWiki/SemanticMediaWiki/pull/5451/commits/f020de1b52ff7ff833fdf0394a3ae81ee40139e2 which removed these types.

File: tests/phpunit/Constraint/Constraints/ShapeConstraintTest.php (hunk 7 above)
Comment (mszabo-wikia) [reply tests/phpunit/Constraint/Constraints/ShapeConstraintTest.php:74]: Yeah, calling `->method()` (via a magic method) without setting up expects would work too.

**Pull request merged**